src/components/fileviewer.js

```diff
@@ -2,84 +2,103 @@ import React, { Component } from 'react';
 
 import * as FetchAPI from '../fetch_data';
 
+/* FileViewer loads a raw file for a given revision from Mozilla's hg web.
+ * It uses test coverage information from Active Data to show coverage
+ * for runnable lines.
+ */
 export default class FileViewerContainer extends Component {
   constructor(props) {
     super(props);
     this.state = {
       appError: undefined,
-      rawFile: undefined,
+      coverage: undefined,
+      parsedFile: [],
     };
+  };
 
+  componentDidMount() {
     const { revision, path } = this.props;
+
     FetchAPI.getRawFile(revision, path)
       .then(response => {
         if (response.status !== 200) {
           console.log('Error status code' + response.status);
           return;
         }
-        response.text().then(file => {
-          this.setState({ rawFile: file.split('\n') })
-        });
-      })
+        response.text()
+      .then((text) => this.setState(() => ({ parsedFile: text.split("\n") })))
```

Comment:
I believe you do not need the parameterless anonymous function in this line

Reply:
I'm not sure if it's redundant but the reason I changed it is because of the idea of this blogpost: https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b